File: frontend/models/message.ts
export enum MessageType {
  USER = "user",
  ASSISTANT = "assistant",
}

interface Message {
  type: MessageType;
  content: string;
  messageId: string;
  threadId: string;
  createdAt: string;
  updatedAt: string;
}

interface Thread {
  threadId: string;
  title: string;
  createdAt: string;
  updatedAt: string;
}

interface ThreadCreate {
  threadId: string;
  title: string;
}

interface ThreadUpdate {
  title: string;
}

interface MessageCreate {
  type: MessageType;
  messageId: string;
  content: string;
  threadId: string;
}

export type { Message, Thread, ThreadCreate, ThreadUpdate, MessageCreate };

File: backend/schemas/message.py
from typing import List, Optional
from pydantic import BaseModel
from datetime import datetime
from enum import Enum

class MessageType(str, Enum):
  user="user"
  assistant="assistant"
  
class ThreadBase(BaseModel):
  title: Optional[str]
  
class ThreadCreate(ThreadBase):
  threadId: str
  title: Optional[str] = None

class ThreadUpdate(BaseModel):
  title: Optional[str]
  
class ThreadDelete(BaseModel):
  threadId: str

class ThreadOut(ThreadBase):
  threadId: str
  title: Optional[str] = None
  createdAt: datetime
  updatedAt: datetime
  
  class Config:
    from_attributes = True

class MessageBase(BaseModel):
  type: MessageType
  content: str
  threadId: str
  
class MessageCreate(MessageBase):
  messageId:str

class MessageUpdate(BaseModel):
  content: str

class MessageOut(MessageBase):
  messageId: str
  createdAt: datetime
  updatedAt: datetime
  
  class Config:
    from_attributes = True
    
ThreadOut.update_forward_refs()
